setup.py

```diff
@@ -24,9 +24,12 @@ def find_version(*file_paths):
 
 
 requires = ['jmespath>=0.7.1,<1.0.0',
-            'python-dateutil>=2.1,<2.7.0',
             'docutils>=0.10']
-
+if sys.version_info <= (2, 7):
```

Comment:
I think this switch only works depending on what version of Python you run `setup.py` with, not the version that you're installing this for.

Shouldn't this be something more like:

```python
requires.extend(["python-dateutil>=2.1,<2.7.0;python_version == '2.6'",
                           "python-dateutil>=2.1,<3.0.0;python_version >= '2.7'"])
```

That way it will also work if you build a wheel on Python 3.6 and install it with python 2.6.

Reply:
@pganssle 
Good call.
But `sys.version_info` is also used below, which should be changed then, too.
IIRC this might require a certain version of setuptools etc (I think checking pytest's setup.py might be good).

Reply:
At `dateutil` we ended up doing [this](https://github.com/dateutil/dateutil/blob/master/setup.py#L15) to ensure a minimum requirement, but it might be possible to use `install_requires`.